game_server.js

```diff
@@ -1,199 +1,18 @@
-	boardEntities[i * width - 6] = BLACK_PIXEL;
-	boardEntities[i * width - 9] = BLACK_PIXEL;
-	boardEntities[i * width - 12] = BLACK_PIXEL;
-	boardEntities[i * width - 11] = BLACK_PIXEL;
-	boardEntities[i * width - 10] = BLACK_PIXEL;
-	boardEntities[i * width - 13] = BLACK_PIXEL;
-	boardEntities[i * width - 14] = BLACK_PIXEL;
-	boardEntities[i * width - 15] = BLACK_PIXEL;
-	boardEntities[i * width - 16] = BLACK_PIXEL;
-}
-
-for (let i = 2; i < 6; i++) {
-	boardEntities[i * width - 1] = BLACK_PIXEL;
-
-	boardEntities[i * width - 5] = resetButton;
-	boardEntities[i * width - 4] = resetButton;
-	boardEntities[i * width - 3] = resetButton;
-	boardEntities[i * width - 2] = resetButton;
-
-	boardEntities[i * width - 6] = BLACK_PIXEL;
-
-	boardEntities[i * width - 10] = randomizeBackgroundButton;
-	boardEntities[i * width - 9] = randomizeBackgroundButton;
-	boardEntities[i * width - 8] = randomizeBackgroundButton;
-	boardEntities[i * width - 7] = randomizeBackgroundButton;
-
-	boardEntities[i * width - 11] = BLACK_PIXEL;
-
-	boardEntities[i * width - 15] = randomizeClientDrawColorButton;
-	boardEntities[i * width - 14] = randomizeClientDrawColorButton;
-	boardEntities[i * width - 13] = randomizeClientDrawColorButton;
-	boardEntities[i * width - 12] = randomizeClientDrawColorButton;
-
-	boardEntities[i * width - 16] = BLACK_PIXEL;
-}
-
-for (let i = 6; i < 7; i++) {
-	boardEntities[i * width - 1] = BLACK_PIXEL;
-	boardEntities[i * width - 4] = BLACK_PIXEL;
-	boardEntities[i * width - 3] = BLACK_PIXEL;
-	boardEntities[i * width - 2] = BLACK_PIXEL;
-	boardEntities[i * width - 5] = BLACK_PIXEL;
-	boardEntities[i * width - 8] = BLACK_PIXEL;
-	boardEntities[i * width - 7] = BLACK_PIXEL;
-	boardEntities[i * width - 6] = BLACK_PIXEL;
-	boardEntities[i * width - 9] = BLACK_PIXEL;
-	boardEntities[i * width - 12] = BLACK_PIXEL;
-	boardEntities[i * width - 11] = BLACK_PIXEL;
-	boardEntities[i * width - 10] = BLACK_PIXEL;
-	boardEntities[i * width - 13] = BLACK_PIXEL;
-	boardEntities[i * width - 14] = BLACK_PIXEL;
-	boardEntities[i * width - 15] = BLACK_PIXEL;
-	boardEntities[i * width - 16] = BLACK_PIXEL;
-}
-
-
-// initial game state
-//
-let newGamePixels = new Uint8ClampedArray(width * height * 4);
-
-const updateGameState = function() {
-	for (let i = 0; i < gamePixels.length; i+=4) {
-		let color = COLORS.BLACK;
-		if ((i/4) in boardEntities) {
-			color = boardEntities[i/4].color;
-		} else {
-			let y = ((i/4) % width);
-			color = changedPixels[(i/4)];
-			if(y < (width / 2)) {
-				color = color ? color : leftSideColor;
-			} else {
-				color = color ? color : rightSideColor;
-			}
-		}
-		newGamePixels[i] = color[0];
-		newGamePixels[i + 1] = color[1];
-		newGamePixels[i + 2] = color[2];
-		newGamePixels[i + 3] = color[3];
-	}
-
-	gamePixels = newGamePixels;
-};
-
-updateGameState();
-
+const draw = new Draw();
+const session = new GameSession(draw, {'width': 320, 'height': 180});
 
 server.on('connection', (ws) => {
 	ws.id = uuid();
```

Comment:
this should exist in `Player`'s constructor, as the WS event handlers are set up there as well.